src/leap/mail/imap/messages.py

```diff
@@ -208,6 +208,15 @@ def getSubPart(self, part):
         return IMAPMessagePart(subpart)
 
 
+class CaseInsensitiveDict(dict):
```

Comment:
Here (or below where is used) it might be useful to have a comment about the rationale of why we need this dict.